system-tests/_support/utils/StringUtil.js

```diff
@@ -0,0 +1,14 @@
+module.exports = {
+
+  /**
+   * Return the element's text contents, convert new lines to spaces and then
+   * trim duplicate whitespaces.
```

Comment:
it doesn't trim duplicate white spaces if `replace` introduces them, only those that were there already. Because `trim` function removes white spaces from the beginning/end of a string.